docs/syntax-key-destructuring.md

```diff
@@ -161,22 +167,48 @@ While with only this RFC, this is unambiguous with 2 lookahead (`{` + `.`), it c
 
 ## Drawbacks
 
+### Only identifiers as keys
+This syntax only proposes `.NAME`, which means it must be a valid name, even if renamed.
+
+The syntax could support `.["key with spaces"]`, but this may open up a can of worms about custom expressions (`.[expr]`). This proposal does not add support for it, though it also does not block it from being added.
+
+This also blocks nested destructuring, such as JavaScript's `const { a: { b } } = t` to mean `b = t.a.b`.
+
 ### Roblox - Property casing
-Today in Roblox, every index doubly works with camel case, such as `part.position` being equivalent to `part.Position`. This use is considered deprecated and frowned upon. However, without (or even with) variable renaming, this becomes significantly more appealing. For example, it is common you will only want a few pieces of informaiton from a `RaycastResult`, so you might be tempted to write:
+Today in Roblox, every index doubly works with camel case, such as `part.position` being equivalent to `part.Position`. This use is considered deprecated and frowned upon. However, even with variable renaming, this becomes significantly more appealing. For example, it is common you will only want a few pieces of informaiton from a `RaycastResult`, so you might be tempted to write:
 
 ```lua
-local { position } = Workspace:Raycast(etc)
+local { .position } = Workspace:Raycast(etc)
 ```
 
 ...which would work as you expect, but rely on this deprecated style.
 
 ## Alternatives
 
-### Syntax
+### Syntax for destructuring
 
 Many syntaxes have been proposed for destructuring. The most significant problem with any proposal is that is must be unambiguous to the reader whether or not the destructor is for **dictionaries** or for **arrays**.
 
 An intuitive suggestion is `local { a, b } = t`, but this syntax fails this test--it is not obvious if this is `local a = t.a` or `local a = t[1]`, regardless of whatever syntax is chosen for array destructuring (should it exist).
 
-### Focus more precisely, rather than using binding
-We could only allow this more precisely, such as just on local assignments. This would satisfy the use case of library importing and React properties (partially, since it would be its own line), but this would limit the use case of tables as arguments to functions, such as in the motivating HTTP request example.
+### Syntax for renaming
+
+`as` does not currently exist in Luau, and thus will not be intuitive to write. We could use `=`, but this doesn't produce a delightful syntax either.
+
+If we simply replace `as` with `=`, though it would parse, it does not read like any other assignment in Luau.
```

Comment:
Note that in JavaScript the `=` has different semantics: if the key is missing from the object, then replace the key by the expression to the right of the `=`.